librz/bin/p/bin_luac.c

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_bin.h>
+#include <rz_lib.h>
+#include "luac/luac_54.h"
+
+static int MAJOR_VERSION;
```

Comment:
I encountered a problem that how to pass this context info to `info` or `check_buffer` functions ? there seems no place to store context info in function arguments or `RzBinFile`.

Reply:
oh. so, check_buffer and load_buffer are used for different purposes.
i understand the issue, in which you fell, because i did the same.

So my suggestion is to leave the `check_buffer` on the version as is, and use `load_buffer` to define a context which is returned in `void **bin_obj`.

For example:
![image](https://user-images.githubusercontent.com/561184/110235871-e1e67000-7f32-11eb-8e7a-86b1e78937d9.png)

Then you can read that structure like this:

![Screenshot_20210307_105114](https://user-images.githubusercontent.com/561184/110235920-283bcf00-7f33-11eb-83a8-2ed33431b529.png)

Reply:
Would this bin_obj be freed by rizin core ? or I should free it somewhere by myself ?